packages/fxa-auth-server/fxa-oauth-server/lib/grant.js

```diff
@@ -169,3 +172,29 @@ function generateIdToken(grant, access) {
 
   return ID_TOKEN_KEY.sign(claims);
 }
+
+function generateAccessToken(id, grant) {
+  var now = Math.floor(Date.now() / 1000);
+
+  const claims = {
+    aud: hex(grant.clientId),
+    auth_time: grant.authAt,
+    client_id: hex(grant.clientId),
+    exp: now + ID_TOKEN_EXPIRATION,
+    iat: now,
+    iss: ID_TOKEN_ISSUER,
+    jti: id,
+    sub: hex(grant.userId),
+  };
+
+  if (grant.amr) {
+    claims.amr = grant.amr;
+  }
+  if (grant.aal) {
+    claims['fxa-aal'] = grant.aal;
+    claims.acr = 'AAL' + grant.aal;
+  }
+
+  // TODO encrypt and sign this thing!
```

Comment:
Is this as simple as `ID_TOKEN_KEY.sign(claims)`, at least for the first version?  AFAICT from [1] above it's fine to use the same key for id tokens and access tokens (inasmuch as the text says that IdPs *may* elect to use different keys for the two types of token, so it also permits that they use the same key).

Reply:
It's [recommended that signing be done using asymmetric keys](https://tools.ietf.org/html/draft-ietf-oauth-access-token-jwt-00#section-4), presumably so clients can validate the JWT themselves. If we allowed clients to validate, encrypting the thing doesn't really provide anything extra except to make the token unintelligible to humans. Clients would need to check the signature, then decrypt with the public key. Worthwhile? Not sure. For v1 if we don't allow clients to validate themselves, perhaps.

Reply:
IIUC `ID_TOKEN_KEY` should already be an asymmetric key.